### PR TITLE
Remote read first class: workflow structure

### PR DESCRIPTION
**Pull request by krajorama**
#### What this PR does

Make place for a remote read roundtripper that would wrap limits and blocking checks.

Update the format of how we log the remote read query matchers in query stats and activity tracker. This will enable having the same string in the logs and also the query blocker.

#### Which issue(s) this PR fixes or relates to

Relates to https://github.com/grafana/mimir-squad/issues/2148 (internal, make remote read first class)

#### Checklist

- [X] Tests updated.
- N/A Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- N/A [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: pkg/frontend/querymiddleware/remote_read.go
Comment (pracucci) [inline pkg/frontend/querymiddleware/remote_read.go:156]: [nit] Since this package contains a bunch of stuff, I would clarify the name of this function:

```suggestion
func remoteReadToMetricsQueryRequest(path string, query *prompb.Query) (MetricsQueryRequest, error) {
```

File: pkg/frontend/querymiddleware/remote_read.go
Comment (krajorama) [inline pkg/frontend/querymiddleware/remote_read.go:139]: note: this might be redundant, probably is, but we can remove later

File: pkg/frontend/querymiddleware/remote_read.go
Comment (pracucci) [inline pkg/frontend/querymiddleware/remote_read.go:144]: You can reuse `util.LabelMatchersToString()`

File: pkg/frontend/querymiddleware/remote_read.go
Comment (pracucci) [inline pkg/frontend/querymiddleware/remote_read.go:139]: [nit] Since they're the matchers of a single query:

```suggestion
func remoteReadQueryMatchersToString(q *prompb.Query) (string, error) {
```

File: pkg/frontend/querymiddleware/remote_read.go
Comment (pracucci) [inline pkg/frontend/querymiddleware/remote_read.go:169]: [nit] Since it implements the "request" interface:

```suggestion
type remoteReadQueryRequest struct {
```

File: pkg/frontend/querymiddleware/remote_read.go
Comment (pracucci) [inline pkg/frontend/querymiddleware/remote_read.go:175]: I typically recommend to do it in the test file. This is just for testing.

File: pkg/frontend/querymiddleware/remote_read.go
Comment (pracucci) [inline pkg/frontend/querymiddleware/remote_read.go:67]: [nit] `error` is the global data type. I suggest to get stick to the `err` convention.
```suggestion
		_, err := handler.Do(req.Context(), metricsRequest)
```

File: pkg/frontend/querymiddleware/remote_read.go
Comment (pracucci) [inline pkg/frontend/querymiddleware/remote_read.go:69]: I don't want to block on this, and we can address it separately, but I'm not 💯 convinced about the `AddDetails()` here. The design smell to me is that it's very easy to forget to use `AddDetails()` instead of the generic `errors.Wrap()`. I would like to make this stuff work with the normal `errors.Wrap()`.

**Review by pracucci — APPROVED** [review]
LGTM. I left few nits I would be glad if you could look at before merging. I would like to discuss with you the `AddDetails()`, but we can do it async after this PR has been merged.

File: pkg/frontend/querymiddleware/remote_read.go
Comment (krajorama) [reply pkg/frontend/querymiddleware/remote_read.go:69]: I wasn't sure that's a good idea as our `Wrap` would behave differently than the standard one (not in terms of the string returned, but the structure), but I'll update in the next PR.

File: pkg/frontend/querymiddleware/remote_read.go
Comment (krajorama) [reply pkg/frontend/querymiddleware/remote_read.go:67]: facepalm

**Pull request merged**
